src/new_recur.jl

```diff
@@ -0,0 +1,105 @@
+
+
+
+##### Helper scan funtion which can likely be put into NNLib. #####
+"""
+  scan
+
+Recreating jax.lax.scan functionality in julia.
+"""
+function scan_full(func, init_carry, xs::AbstractVector{<:AbstractArray})
+  # xs = Flux.eachlastdim(x_block)
+  x_init, x_rest = Iterators.peel(xs)
+
+  (carry, out_) = func(init_carry, x_init)
+
+  init = (typeof(out_)[out_], carry)
+
+  function recurrence_op(input, (outputs, carry))
+    carry, out = func(carry, input)
+    return vcat(outputs, typeof(out)[out]), carry
+  end
+  results = foldr(recurrence_op, xs[(begin+1):end]; init)
+  results[2], results[1]
```

Comment:
Right. I got the original impl from Lux.jl, but they have since changed to foldl (and then to their own implementation). I have to dig back into this to see whats going on. You are correct this needs to be foldl.

Reply:
I've fixed this in my new commit. Required some subtle change in the rest of the function and I combined it with the change to the return orders to be consistent w/ the current recurrent cell implementation so it is less confusing.